src/common/util/navigator_gpu.ts

```diff
@@ -58,14 +64,16 @@ export function getGPU(): GPU {
     impl.requestAdapter = function (
       options?: GPURequestAdapterOptions
     ): Promise<GPUAdapter | null> {
-      const promise = oldFn.call(this, { ...defaultRequestAdapterOptions, ...(options || {}) });
-      void promise.then(async adapter => {
-        if (adapter) {
-          const info = await adapter.requestAdapterInfo();
-          // eslint-disable-next-line no-console
-          console.log(info);
-        }
-      });
+      const promise = oldFn.call(this, { ...defaultRequestAdapterOptions, ...options });
+      if (recorder) {
+        void promise.then(async adapter => {
+          if (adapter) {
+            const info = await adapter.requestAdapterInfo();
+            const infoString = `Adapter: ${info.vendor} / ${info.architecture} / ${info.device}`;
+            recorder.debug(new ErrorWithExtra(infoString, () => ({ adapterInfo: info })));
```

Comment:
Instead of falling back to `console.log`ing I've just wired this up to the TestCaseRecorder, since it's the only way to get logs out in WPT anyway.

WDYT of making it a `debug` level log? That means it'll only show up if you specify `?debug=1` in standalone (and not at all in WPT, and at least not by default in Chromium's harness)

Reply:
sgtm